src/components/dialog/dialog.component.tsx

```diff
@@ -37,7 +37,9 @@ export interface ContentPaddingInterface {
   px?: PaddingValues;
 }
 
-export interface DialogProps extends ModalProps, TagProps {
+export interface DialogProps
+  extends ModalProps,
+    Omit<TagProps, "data-component"> {
```

Comment:
Yeah, I debated about flagging this too. I'm going to flag this as a breaking change, since there's a chance a consumer may be querying the root element of the `Modal` in their UI tests